mypyc/test-data/run-functions.test

```diff
@@ -1256,3 +1256,33 @@ def foo(**kwargs: Unpack[Person]) -> None:
 foo(name='Jennifer', age=38)
 [out]
 Jennifer
+
+[case testNestedFunctionDunderDict312]
+import sys
+
+def foo() -> None:
+    def inner() -> str: return "bar"
+    print(inner.__dict__)  # type: ignore[attr-defined]
```

Comment:
There's a lot of type ignores in this test, do we really need that?

Reply:
yes, fixtures